src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -2,34 +2,65 @@
 
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Diagnostics;
 using System.Reflection;
+using System.Threading;
 using Microsoft.Cci;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Roslyn.Utilities;
-using System.Diagnostics;
-using System.Threading;
-using System;
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal class LocalFunctionSymbol : MethodSymbol
     {
+        private sealed class ParametersAndDiagnostics
+        {
+            internal readonly ImmutableArray<ParameterSymbol> Parameters;
+            internal readonly bool IsVararg;
+            internal readonly ImmutableArray<Diagnostic> Diagnostics;
+
+            internal ParametersAndDiagnostics(ImmutableArray<ParameterSymbol> parameters, bool isVararg, ImmutableArray<Diagnostic> diagnostics)
+            {
+                Parameters = parameters;
+                IsVararg = isVararg;
+                Diagnostics = diagnostics;
+            }
+        }
+
+        private sealed class TypeParameterConstraintsAndDiagnostics
+        {
+            internal readonly ImmutableArray<TypeParameterConstraintClause> ConstraintClauses;
+            internal readonly ImmutableArray<Diagnostic> Diagnostics;
+
+            internal TypeParameterConstraintsAndDiagnostics(ImmutableArray<TypeParameterConstraintClause> constraintClauses, ImmutableArray<Diagnostic> diagnostics)
+            {
+                ConstraintClauses = constraintClauses;
+                Diagnostics = diagnostics;
+            }
+        }
+
+        private sealed class ReturnTypeAndDiagnostics
+        {
+            internal readonly TypeSymbol ReturnType;
+            internal readonly ImmutableArray<Diagnostic> Diagnostics;
+
+            internal ReturnTypeAndDiagnostics(TypeSymbol returnType, ImmutableArray<Diagnostic> diagnostics)
+            {
+                ReturnType = returnType;
+                Diagnostics = diagnostics;
+            }
+        }
+
         private readonly Binder _binder;
         private readonly LocalFunctionStatementSyntax _syntax;
         private readonly Symbol _containingSymbol;
         private readonly DeclarationModifiers _declarationModifiers;
         private readonly ImmutableArray<TypeParameterSymbol> _typeParameters;
-        private ImmutableArray<ParameterSymbol> _parameters;
-        private ImmutableArray<TypeParameterConstraintClause> _lazyTypeParameterConstraints;
         private readonly RefKind _refKind;
-        private TypeSymbol _returnType;
-        private bool _isVararg;
+        private ParametersAndDiagnostics _lazyParametersAndDiagnostics;
+        private TypeParameterConstraintsAndDiagnostics _lazyTypeParameterConstraintsAndDiagnostics;
+        private ReturnTypeAndDiagnostics _lazyReturnTypeAndDiagnostics;
         private TypeSymbol _iteratorElementType;
-
-        // TODO: Find a better way to report diagnostics.
```

Comment:
The change was needed in an earlier iteration of this PR but is no longer strictly necessary.